flow/scripts/floorplan.tcl

```diff
@@ -87,6 +87,34 @@ if {[info exists ::env(MACRO_WRAPPERS)]} {
   }
 }
 
+##### Restructure for timing #########
+if { [info exist ::env(RESYNTH_TIMING_RECOVER)] && $::env(RESYNTH_TIMING_RECOVER) == 1 } {
```

Comment:
Its a bit odd to have both RESYNTH_TIMING_RECOVER and RESYNTH_AREA_RECOVER since they seem mutually exclusive.  Would it be better to have RESYNTH_MODE with two values?

Reply:
Technically you can do both but I have not tried it. Single variable would not allow anyone to try both together if one wants to. This code is to allow experimentation of these modes. Ideally using this infrastructure a optimization recipe need to be arrived at which would be a separate project. For example, timing remap could be done post area remap as well.
All this code can be removed as well since this is primarily to show how this features can be used. Eventually, this could all go in timing driven placement mega command.